File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        int count = 0;
        long n = num;
        if(num == 1){
            return answer;
        }
        while(n > 1){
            if(count >= 500){
                break;
            }
            if(n % 2 == 0){
                n = n / 2;
            }else{
                n = (n * 3) + 1;
            }
            count++;
        }
        if(count >= 500){
            answer = -1;
        }else{
            answer = count;
        }

        return answer;
    }
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.swift
func solution(_ num:Int) -> Int {
    var result = 0
    var count = 0
    var n:Int64 = Int64(num)
    if n == 1 {
        return result
    }
    while(n > 1) {
        if(count >= 500){
            break
        }
        if(n % 2 == 0){
            n /= 2
        } else {
            n = (n * 3) + 1
        }
        count += 1
    }
    if(count >= 500){
        result = -1
    } else {
        result = count
    }
    return result
}